core/src/main/scala/schema/provider/MetadataSchemaProvider.scala

```diff
@@ -52,14 +58,15 @@ class MetadataSchemaProvider[M[_]](cluster: Cluster)(implicit H: FSHandler[Clust
         M.flatMap(finalizer)(_ => e.fold(M.raiseError, M.pure))
```

Comment:
I know that's unrelated but in this scope, we have `M` as MonadError evidence, which matches with the class type constructor `M[_]`. Maybe we could rename it, wdyt?